pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -1558,8 +1565,62 @@ func (volTopology *wcpControllerVolumeTopology) GetTopologyInfoFromNodes(ctx con
 
 	switch strings.ToLower(params.StorageTopologyType) {
 	case "zonal":
-		// If the topology requirement received has just one zone, use the same zone as node affinity terms on PV.
-		if len(params.TopologyRequirement.GetPreferred()) == 1 {
+		if params.TopologyRequirement == nil {
+			// If the topology requirement received is nil, then identigy topology of the datastore by looking into
+			// azClustersMap
+			if !isPodVMOnStretchedSupervisorEnabled {
+				return nil, logger.LogNewErrorf(log, "invalid params: %v", params)
```

Comment:
Could we elaborate the error message as `invalid volume creation request. Parameters: %+v`

Reply:
done

Reply:
not printing invalid volume creation request as we do not expect user to supply this information in the CNSRegisterVolume API spec. but instead printing this log

```
return nil, logger.LogNewErrorf(log, "topology requirement should not be nil. invalid params: %v", params)
```